__test__/main.test.ts

```diff
@@ -6,26 +6,97 @@ import * as osx from "../src/setup-ros-osx";
 import * as utils from "../src/utils";
 
-utils.lib.exec = jest.fn(async (commandLine: string,
-    args?: string[],
-    options?: im.ExecOptions,
-    log_message?: string) => 
-    {
-        const argsAsString = (args || []).join(" ");
-        const message = log_message || `Invoking "${commandLine} ${argsAsString}"`;
-        return core.group(message, async () => {
-            return 0;
-        });
-    });
-
-test('run Windows workflow', async () => {
-    return windows.runWindows();
-})
 
-test('run Linux workflow', async () => {
-    return linux.runLinux();
-})
+describe('basic workflow tests', () => {
+    let linesOfExec: string[] = [];
+    beforeAll(() => {
+        jest
+            .spyOn(utils, 'exec')
+            .mockImplementation(async (
+                commandLine: string,
+                args?: string[],
+                options?: im.ExecOptions,
+                log_message?: string) => {
+                    let flatArgs = ''
+                    if (args) {
+                        flatArgs = args.map(i => i).join(' ')
+                    }
+                    linesOfExec.push(`${commandLine} ${flatArgs}`)
+                    return 0;
+                });
+    })
+
+    beforeEach(() => {
+        linesOfExec = [];
+    })
+
+    afterAll(() => {
+        jest.restoreAllMocks();
+    })
 
-test('run macOS workflow', async () => {
-    return osx.runOsX();
+    it('run Linux workflow', async () => {
+        await linux.runLinux();
+        console.log(linesOfExec);
+    })
+
+    it('run Windows workflow', async () => {
+        await windows.runWindows();
+        console.log(linesOfExec);
+    })
+
+    it('run macOS workflow', async () => {
+        await osx.runOsX();
+        console.log(linesOfExec);
+    })
 })
+
+describe('required-ros-distributions workflow tests', () => {
+    let linesOfExec: string[] = [];
+
+    beforeAll(() => {
+        jest
+            .spyOn(utils, 'exec')
+            .mockImplementation(async (
+                commandLine: string,
+                args?: string[],
+                options?: im.ExecOptions,
+                log_message?: string) => {
+                    let flatArgs = ''
+                    if (args) {
+                        flatArgs = args.map(i => i).join(' ')
+                    }
+                    linesOfExec.push(`${commandLine} ${flatArgs}`)
+                    return 0;
+                });
+        jest
+            .spyOn(core, 'getInput')
+            .mockReturnValue('melodic')
+    })
+
+    beforeEach(() => {
+        linesOfExec = [];
+    })
+
+    afterAll(() => {
+        jest.restoreAllMocks();
+    })
+
+    it('run Linux workflow', async () => {
+        await linux.runLinux();
+        console.log(linesOfExec);
+        expect(linesOfExec)
+            .toContain('sudo DEBIAN_FRONTEND=noninteractive RTI_NC_LICENSE_ACCEPTED=yes apt-get install --no-install-recommends --quiet --yes ros-melodic-desktop')

```

Comment:
Any reason for this line specifically? (Same with windows but not with mac?)

Reply:
I was adding some ad-hoc validation but I agreed with you that we can just do `.not.toThrow().` for now.